### PR TITLE
Initial v2 api

### PR DESCRIPTION
**Pull request by croomes**
Uses v2 directory as recommended in https://blog.golang.org/v2-go-modules

Allows multiple versions to be imported as:
```
import (
        apiv2 "github.com/storageos/go-api/v2"
        apiv1 "github.com/storageos/go-api"
)
```
With the exception of `v2/Makefile` and `v2/openapi.yaml`, all other files in `v2/` are generated by https://github.com/OpenAPITools/openapi-generator.

It may be possible to improve the generated `README.md`, but I haven't looked closely yet.

### CODE REVIEW

**Comment by darkowlzz** [comment]
CI trigger seems to be failing. I'm seeing `storageos/go-api has no workflows configured` at https://circleci.com/gh/storageos/workflows/go-api.
Maybe adding
```
workflows:
  version: 2
  build_and_test:
    jobs:
      - build
```
to [.circleci/config.yml](https://github.com/storageos/go-api/blob/master/.circleci/config.yml) (as per https://circleci.com/docs/2.0/workflows/) would fix the CI trigger.

File: v2/Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+.DEFAULT_GOAL:=help
+
+.PHONY: help customise
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n\nTargets:\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-13s\033[0m %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
+build: generate customise ## Re-generates the API from openapi.yaml
+
+generate:
```
Comment (darkowlzz) [inline v2/Makefile]: `##` comment for this to appear in the make list of commands.

**Comment by darkowlzz** [comment]
I tried generating it myself and it works well. Need to regenerate with `storageos` name before we can merge it.

File: v2/Makefile (hunk 1 above)
Comment (croomes) [reply v2/Makefile]: done

File: v2/Makefile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		openapitools/openapi-generator-cli:latest generate \
+		--strict-spec=true \
+		--additional-properties enumClassPrefix=true \
+		--additional-properties packageName=api \
+		--additional-properties packageVersion=$(VERSION) \
+		-i /local/openapi.yaml -g go -o /local
+
+customise:
+	sed -i 's/\[]map/map/ig' *.go
+	sed -i "s#module github.com/GIT_USER_ID/GIT_REPO_ID#module github.com/${GIT_USER_ID}/${GIT_REPO_ID}/v2#g" go.mod
```
[the remark this answers is not in the document]
Comment (croomes) [reply v2/Makefile]: done

File: v2/go.mod [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module github.com/croomes/go-api/v2
```
[the remark this answers is not in the document]
Comment (croomes) [reply v2/go.mod]: done

File: v2/.travis.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+language: go
+
+install:
+  - go get -d -v .
+
+script:
+  - go build -v ./
+
```
[the remark this answers is not in the document]
Comment (croomes) [reply v2/.travis.yml]: yes, not sure how useful it is - we can delete it for now in the customise step?

[the remark this answers is not in the document]
Comment (darkowlzz) [reply v2/.travis.yml]: No problem in keeping it as it is, since it's in a subdirectory. We can figure out later how to disable generating this.

**Pull request merged**